### PR TITLE
63 improve qmake and deps

### LINKED ISSUE #63: Improve qmake OS detection on Linux
Right now qmake is only checking if we are on Windows or Linux. Since different dependencies are available on different Linux flavors, checking for the distribution we could avoid compiling unnecessary packages.

To check the distro in qmake something like this could work:

```
ARCH = $$system(cat /proc/version | grep -o Arch) 
UBUNTU = $$system(cat /proc/version | grep -o Ubuntu) 

contains( ARCH, Arch): 
{     
   do stuff
} 
contains( UBUNTU, Ubuntu): 
{     
   do other stuff
}
```

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes #63

### CODE REVIEW

File: Cascade.pro [hunk 1]
```diff
@@ -287,31 +287,47 @@ FORMS += \
...
+    -lSPIRV-Tools \
+    -lMachineIndependent \
+    -lglslang \
+    -lglslang-default-resource-limits \
+    -lOSDependent \
+    -lOGLCompiler \
+    -lGenericCodeGen
+
+    LIBS += -L/usr/lib/x86_64-linux-gnu -ldl -ltbb
     # The link order of the following libs is important
```
Comment (ttddee) [inline Cascade.pro:329]: This comment needs to go above the SPIRV includes

**Review by ttddee — COMMENTED** [review]
Testing on EndeavourOS the check fails, the output of `uname -a` is:

`Linux arch 5.18.14-arch1-1`

Maybe we can add another variable that checks for the string `arch`, so that it works on different flavors of Arch?

File: Cascade.pro (hunk 1 above)
Comment (Yaraslaut) [reply Cascade.pro]: Corrected

Comment (ttddee) [reply Cascade.pro]: Sorry I meant adding a check to the pro file. Right now we are only testing if we are on Manjaro with this:

`isManjaro = $$find(OS,Manjaro)`

We could generalize that to other Arch distros with something like this:

`isArch = $$find(OS,arch)`

What do you think?

Comment (Yaraslaut) [reply Cascade.pro]: yes, it also makes sence, updated

**Pull request merged**